ethpm_types/manifest.py

```diff
@@ -46,33 +50,96 @@ def __repr__(self):
 
 
 class PackageMeta(BaseModel):
+    """
+    Important data that is not integral to installation
+    but should be included when publishing.
+    """
+
     authors: Optional[List[str]] = None
+    """A list of human readable names for the authors of this package."""
+
     license: Optional[str] = None
+    """
+    The license associated with this package.
+    This value should conform to the SPDX format.
+    Packages should include this field.
+    If a file Source Object defines its own license, that license takes
+    precedence for that particular file over this package-scoped meta license.
+    """
+
     description: Optional[str] = None
+    """Additional detail that may be relevant for this package."""
+
     keywords: Optional[List[str]] = None
+    """Relevant keywords related to this package."""
+
     links: Optional[Dict[str, AnyUrl]] = None
+    """
+    URIs to relevant resources associated with this package.
+    When possible, authors should use the following keys for the following common resources.
+    """
 
 
 class PackageManifest(BaseModel):
+    """
+    A data format describing a smart contract software package.
+
+    `EIP-2678<https://eips.ethereum.org/EIPS/eip-2678#ethpm-manifest-version>`__
+    """
+
     manifest: str = "ethpm/3"
+    """The specification version that the project conforms to."""
+
     name: Optional[PackageName] = None
-    # NOTE: ``version`` should be valid SemVer
+    """A human-readable name for the package."""
+
     version: Optional[str] = None
-    # NOTE: ``meta`` should be in all published packages
+    """The version of the release, which should be SemVer."""
+
     meta: Optional[PackageMeta] = None
-    # NOTE: ``sources`` source tree should be necessary and sufficient to compile
-    #       all ``ContractType``s in manifest
+    """
+    Important data that is not integral to installation
+    but should be included when publishing.
```

Comment:
The note was removed from above, but add the recommendation that all published packages should have the meta field

Reply:
Ah, I tried when by saying `but should be included when publishing` but I will adjust it